### PR TITLE
Fix bugs

### LINKED ISSUE #173: [PE-D][Tester D] Misleading output printed
When adding NOK to doctor, output is misleading. User does not know what they are inputting wrongly.

Write something like "NOK only allowed to be added to patient"

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: ryanongwx/ped#1

### LINKED ISSUE #183: [PE-D][Tester D] NOK feature flaw
As shown in the user guide, it states that when i call nok 2 without any inputs it will delete the nok. But when i call nok 2 without any inputs on a valid patient, the output is Next of Kin added! as shown in first image. But no next of kin added

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: ryanongwx/ped#3

### LINKED ISSUE #189: [PE-D][Tester A] No data upon booting the application up again
As shown in the picture below, when i exit and tried to run the program again, there is nothing shown. I have the `medisync.json` file in the `data` directory, hence there seems to be a problem when reading the json file. I understadn that under your `Editing the data file` section, there seem to have a lot of cautions. I assure that I did not directly modify the json file, and all I did was to run the commands in the application. Hence I think it would be good if you can look into why is this causing a problem where the commands that changes the file could lead to this.
The `medisync.json` file is as follows:
```
{
  "persons" : [ {
    "name" : "Alex Yeoh",
    "phone" : "87438807",
    "email" : "alexyeoh@example.com",
    "role" : "Doctor",
    "shiftDays" : [ ],
    "specialisations" : [ "Radiologist", "ENT" ],
    "affiliations" : [ "Bernice Yu", "Charlotte Oliveiro" ],
    "affiliationHistory" : [ "Bernice Yu", "Charlotte Oliveiro" ]
  }, {
    "name" : "Bernice Yu",
    "phone" : "99272758",
    "email" : "berniceyu@example.com",
    "role" : "Patient",
    "nextOfKin" : {
      "nextOfKinName" : "Holland Tan",
      "nextOfKinPhone" : "91334428",
      "nextOfKinRelationship" : "Father"
    },
    "affiliations" : [ "Alex Yeoh" ],
    "affiliationHistory" : [ "Alex Yeoh" ]
  }, {
    "name" : "May Ho",
    "phone" : "94437233",
    "email" : "homimay@example.com",
    "role" : "Nurse",
    "shiftDays" : [ 3, 4, 5 ],
    "affiliations" : [ "Irfan Ibrahim" ],
    "affiliationHistory" : [ ]
  }, {
    "name" : "Charlotte Oliveiro",
    "phone" : "93210283",
    "email" : "charlotte@example.com",
    "role" : "Patient",
    "nextOfKin" : {
      "nextOfKinName" : "Ancob Maximus",
      "nextOfKinPhone" : "87769988",
      "nextOfKinRelationship" : "Husband"
    },
    "affiliations" : [ "Alex Yeoh", "Evelyn Ng" ],
    "affiliationHistory" : [ "Alex Yeoh", "Evelyn Ng" ]
  }, {
    "name" : "David Li",
    "phone" : "91031282",
    "email" : "lidavid@example.com",
    "role" : "Doctor",
    "shiftDays" : [ 2, 5, 7 ],
    "specialisations" : [ "heart brain cell" ],
    "affiliations" : [ "Irfan Ibrahim", "Testing2" ],
    "affiliationHistory" : [ "Irfan Ibrahim", "Testing2" ]
  }, {
    "name" : "Irfan Ibrahim",
    "phone" : "92492021",
    "email" : "irfan@example.com",
    "role" : "Patient",
    "nextOfKin" : {
      "nextOfKinName" : null,
      "nextOfKinPhone" : null,
      "nextOfKinRelationship" : null
    },
    "affiliations" : [ "May Ho", "David Li" ],
    "affiliationHistory" : [ "May Ho", "David Li" ]
  }, {
    "name" : "Evelyn Ng",
    "phone" : "92624417",
    "email" : "eveng@example.com",
    "role" : "Nurse",
    "shiftDays" : [ 1, 2, 4, 6 ],
    "affiliations" : [ "Charlotte Oliveiro" ],
    "affiliationHistory" : [ "Charlotte Oliveiro" ]
  }, {
    "name" : "Testing2",
    "phone" : "987654",
    "email" : "fjieigj@example.com",
    "role" : "Patient",
    "nextOfKin" : {
      "nextOfKinName" : null,
      "nextOfKinPhone" : null,
      "nextOfKinRelationship" : null
    },
    "affiliations" : [ "David Li" ],
    "affiliationHistory" : [ ]
  }, {
    "name" : "Testing3",
    "phone" : "93874947",
    "email" : "firjig@example.com",
    "role" : "Patient",
    "nextOfKin" : {
      "nextOfKinName" : null,
      "nextOfKinPhone" : null,
      "nextOfKinRelationship" : null
    },
    "affiliations" : [ ],
    "affiliationHistory" : [ ]
  } ]
}
```

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: brendanneojw/ped#11

### LINKED ISSUE #191: [PE-D][Tester C] Cosmetic issue when editing contact details
Perhaps the error message could show that you cannot affiliate a person with themselves?

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: jordankanghm/ped#1

### PR DESCRIPTION
**Pull request by Seonlo99**
Fixes #173, #183, #189, #191

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public class Messages {
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
     public static final String MESSAGE_DUPLICATE_FIELDS =
                 "Multiple values specified for the following single-valued field(s): ";
-    public static final String MESSAGE_INVALID_ROLE = "Invalid role!";
+    public static final String MESSAGE_INVALID_ROLE = "Next-of-Kin are not available for Doctors and Nurses!";
```
Comment (leezhanpeng) [inline src/main/java/seedu/address/logic/Messages.java]: If I recall, some other commands also use this message. Will the change tally with them?

**Review by leezhanpeng — COMMENTED** [review]
One of the message changes might be problematic. Also, would be great if you are able to match the associated bug issues to this PR.

File: src/main/java/seedu/address/logic/Messages.java (hunk 1 above)
Comment (Seonlo99) [reply src/main/java/seedu/address/logic/Messages.java]: You are right! Thanks for pointing it out will change it in the next commit.

**Review by leezhanpeng — APPROVED** [review]
LGTM. This fixes a critical bug whereby running of `removeah` command will cause the affiliation history of a person to be deleted wrongly, resulting in invalid data. If the user restarts MediSync, the data will be reset and cleared, which should not be the case.

**Pull request merged**